oci/spec_opts_linux.go

```diff
@@ -82,9 +83,9 @@ func getDevices(path string) ([]specs.LinuxDevice, error) {
 	return out, nil
 }
 
-func deviceFromPath(path, permissions string) (*specs.LinuxDevice, error) {
+func deviceFromPath(hostPath, containerPath, permissions string) (*specs.LinuxDevice, error) {
```

Comment:
CI failures are because you changed the signature for the linux variant, but not the "unix" variant (which is built on macOS), and Windows variant;

https://github.com/containerd/containerd/blob/08d765af954cd5355b7207fe628d06ed5a97c9d4/oci/spec_opts_unix.go#L85

https://github.com/containerd/containerd/blob/d8527867057daa93c17423ec7cd3438696b1c899/oci/spec_opts_windows.go#L77

Reply:
So I have to keep deviceFromPath signature same for all platforms?

Reply:
Yes, they're platform-specific implementations of the same function, so must have the same signature (on Linux, calling `deviceFromPath` will use `oci/spec_opts_linux.go`, on Windows, it will use `oci/spec_opts_windows.go`, and on other non-linux platforms it uses `oci/spec_opts_unix.go`

Reply:
(I should add that `deviceFromPath()` from the `_windows.go` file is not used in this case, but it's probably good to look if makes sense to keep the signature the same)

Reply:
Sorry I was busy last week and I am resuming working on this.

I see your point and I will update all of them.